app/scripts/metamask-controller.js

```diff
@@ -741,6 +749,15 @@ export default class MetamaskController extends EventEmitter {
       ///: END:ONLY_INCLUDE_IN
     }
 
+    const keyringOverrides = this.opts.overrides?.keyrings;
+    additionalKeyrings = additionalKeyrings.map((keyring) => {
+      const keyringOverride = keyringOverrides?.find(
+        (override) => override.type === keyring.type,
+      );
+
+      return keyringOverride || keyring;
+    });
```

Comment:
It's going to be easier to override the keyrings this way, as we still haven't written bridges for trezor and ledger for desktop.